### PR TITLE
Installer script

### PR DESCRIPTION
**Pull request by markuskowa**
Draft for the installer.
Setup tests with 
* [x] Arch Linux
* [x] Ubuntu
* [x] Debian
* [ ] Fedora

### CODE REVIEW

**Comment by sheepforce** [comment]
Works on Debian Bullseye.

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+EOF
+else
+  printf "Config file $config_file exists. Skipping config creation."
+fi
+
+
+printf "\n\n"
+printf "The Nix and the NixOS-QChem overlay are now\n"
+printf "installed in you system. Before you can use it\n"
+printf "you need refresh your by launching a fresh shell or doing re-login'\n"
```
Comment (sheepforce) [inline install.sh:126]: New shell and sourcing the bashrc is not enough at least in Debian and Ubuntu.

**Review by sheepforce — CHANGES_REQUESTED** [review]

**Comment by sheepforce** [comment]
Works on Ubuntu 20.04 LTS.

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+EOF
+
+  exit 0
+fi
+
+# Check if git is available
+command -v git --version > /dev/null || error "Git is not installed. We need git to check out the overlay."
+
+# Needed by the nix installer
+command -v curl --version > /dev/null || error "Curl is not installed."
```
[the remark this answers is not in the document]
Comment (markuskowa) [reply install.sh:34]: I added a new line to the error message.

File: install.sh (hunk 1 above)
Comment (markuskowa) [reply install.sh]: Yes, the issue is that the PATH is updated via `/etc/profile` and not `bashrc`. What you need is a fresh login shell.

**Comment by sheepforce** [comment]
Fedora does NOT work with a minor linking error in the user environment
[nix-install.log](https://github.com/markuskowa/NixOS-QChem/files/7286410/nix-install.log)
:

**Comment by markuskowa** [comment]
> Fedora does NOT work with a minor linking error in the user environment [nix-install.log](https://github.com/markuskowa/NixOS-QChem/files/7286410/nix-install.log) :

Bummer, but not our fault as it is the nix installer quitting (not clear to me why this happens).

**Pull request merged**